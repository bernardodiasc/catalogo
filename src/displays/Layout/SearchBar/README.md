# SearchBar

## How to use

```
import SearchBar from 'displays/Layout/SearchBar'
```

```
<SearchBar />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
