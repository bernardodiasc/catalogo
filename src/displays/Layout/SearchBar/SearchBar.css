.Layout-SearchBar {
  display: flex;
  justify-content: center;
  width: 100%;
  padding: 10px 0;
}

.Layout-SearchBar__selectbox {
  width: 250px;
  padding: 0 10px;
}
