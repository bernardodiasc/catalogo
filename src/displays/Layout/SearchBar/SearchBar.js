import React, { PureComponent } from 'react'
import './SearchBar.css'

import SelectBox from 'displays/SelectBox'

class SearchBar extends PureComponent {
  static defaultProps = {
    onSelect: () => {},
  }

  handleChange = (selectedOption) => {
    this.props.onSelect(selectedOption.value)
  }

  render() {
    return (
      <div className="Layout-SearchBar">
        <div className="Layout-SearchBar__selectbox">
          <SelectBox
            name="select-city"
            placeholder="Selecione uma cidade..."
            onChange={this.handleChange}
            options={[
              { value: 'ubatuba', label: 'Ubatuba' },
            ]}
          />
        </div>
        <div className="Layout-SearchBar__selectbox">
          <SelectBox
            name="select-category"
            placeholder="Selecione uma categoria..."
            onChange={this.handleChange}
            options={[
              { value: 'escola', label: 'Escola' },
              { value: 'padaria', label: 'Padaria' },
            ]}
          />
        </div>
      </div>
    )
  }
}

export default SearchBar
