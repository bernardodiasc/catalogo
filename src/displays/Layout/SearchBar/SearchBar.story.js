import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import Docs from 'stories/containers/Docs'

import SearchBar from './SearchBar'
import SearchBarReadme from './README.md'
storiesOf('Displays/Layout/SearchBar', module)
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{SearchBarReadme}</Docs>)
  .add('Example', () => <SearchBar />)
  .add('Minimal render', () => <SearchBar />)
