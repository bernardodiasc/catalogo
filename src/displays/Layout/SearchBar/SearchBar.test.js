import React from 'react'
import { shallow } from 'enzyme'

import SearchBar from './SearchBar.js'

it('renders without crashing', () => {
  const component = shallow(<SearchBar />)
  expect(component).toMatchSnapshot()
})
