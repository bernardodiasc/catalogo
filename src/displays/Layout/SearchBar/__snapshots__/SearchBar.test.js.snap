// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Layout-SearchBar"
>
  <div
    className="Layout-SearchBar__selectbox"
  >
    <SelectBox
      name="select-city"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "Ubatuba",
            "value": "ubatuba",
          },
        ]
      }
      placeholder="Selecione uma cidade..."
    />
  </div>
  <div
    className="Layout-SearchBar__selectbox"
  >
    <SelectBox
      name="select-category"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "Escola",
            "value": "escola",
          },
          Object {
            "label": "Padaria",
            "value": "padaria",
          },
        ]
      }
      placeholder="Selecione uma categoria..."
    />
  </div>
</div>
`;
